server/api/documents.js

```diff
@@ -621,36 +645,48 @@ router.post('documents.update', auth(), async ctx => {
 
   // Update document
   if (title) document.title = title;
-  //append to document
+
   if (append) {
     document.text += text;
   } else if (text) {
     document.text = text;
   }
   document.lastModifiedById = user.id;
 
-  if (publish) {
-    await document.publish();
-
-    events.add({
-      name: 'documents.publish',
-      modelId: document.id,
-      collectionId: document.collectionId,
-      teamId: document.teamId,
-      actorId: user.id,
-    });
-  } else {
-    await document.save({ autosave });
-
-    events.add({
-      name: 'documents.update',
-      modelId: document.id,
-      collectionId: document.collectionId,
-      teamId: document.teamId,
-      actorId: user.id,
-      autosave,
-      done,
-    });
+  let transaction;
+  try {
+    transaction = await sequelize.transaction();
```

Comment:
New transaction is required to ensure that the revision is saved as part of document creation transaction and can be guaranteed to be available on the worker